src/hybrid_failure_discovery/envs/conveyorbelt_env.py

```
@@ -0,0 +1,247 @@
+
+    def __init__(self, render_mode: str = "rgb_array", seed: int = 0) -> None:
+        self.render_mode = render_mode
+        self.observation_dim = 6
+        self._last_inserted_value: float | None = None
+        self.action_dim = 3  # Now only actions 0, 1, 2 are valid
+        super().__init__(seed)
+
+        self._step_count = 0
+
+    def _create_action_space(self):
+        return range(self.action_dim)
+
+    def _get_obs(self) -> ConveyorBeltState:
+        assert self._current_state is not None
+        return self._current_state
+
+    def get_initial_states(self) -> EnumSpace[ConveyorBeltState]:
+        assert self._np_random is not None
+        init_values = self._np_random.random(self.observation_dim).astype(np.float32)
+        init_state = ConveyorBeltState(values=init_values)
+        return EnumSpace([init_state])
+
+    def get_next_states(
+        self, state: ConveyorBeltState, action: ConveyorBeltAction
+    ) -> EnumSpace[ConveyorBeltState]:
+        # print(f"[DEBUG] Action taken: {action.index}")
+
+        next_values = state.values.copy()
+        assert self._np_random is not None
+
+        if action.index == 0:
+            # print("[DEBUG] No-op: Do nothing")
+            new_value = None
+
+        elif action.index == 1:
+            new_value = float(self._np_random.random())
+            # print("[DEBUG] Shift left and load new box on the right")
+            next_values[:-1] = next_values[1:]
+            next_values[-1] = new_value
+            # print(f"[DEBUG] Loaded new value: {new_value:.2f}")
+
+        elif action.index == 2:
+            new_value = float(self._np_random.random())
+            # print("[DEBUG] Shift right and load new box on the left")
+            next_values[1:] = next_values[:-1]
+            next_values[0] = new_value
+            # print(f"[DEBUG] Loaded new value: {new_value:.2f}")
+
+        else:
+            print(f"[WARNING] Invalid action index: {action.index}")
```

we should probably throw an error here instead of just a warning